client/core/context.go

```diff
@@ -48,3 +52,13 @@ func (c CoreContext) WithClient(client rpcclient.Client) CoreContext {
 	c.Client = client
 	return c
 }
+
+func (c CoreContext) WithDecoder(decoder sdk.AccountDecoder) CoreContext {
```

Comment:
Please add comment headers - what's dis stuff doing anyways

Reply:
Makes it easier to change context parameters one-at-a-time (the IBC module uses these functions quite a bit, for example).

Comment headers added.